pykafka/producer.py

```diff
@@ -237,24 +238,27 @@ def produce(self, message, partition_key=None):
             raise ProducerStoppedException()
         partitions = list(self._topic.partitions.values())
         partition_id = self._partitioner(partitions, partition_key).id
-        message_partition_tup = (partition_key, message), partition_id, 0
-        self._produce(message_partition_tup)
-        if self._synchronous:
-            self._wait_all()
+        msg = Message(value=message,
+                      partition_key=partition_key,
+                      partition_id=partition_id)
+        msg.delivery_future = futures.Future()
+        self._produce(msg)
         self._raise_worker_exceptions()
+        if self._synchronous:
+            return msg.delivery_future.result()
+        return msg.delivery_future
 
-    def _produce(self, message_partition_tup):
+    def _produce(self, message):
         """Enqueue a message for the relevant broker
 
-        :param message_partition_tup: Message with partition assigned.
-        :type message_partition_tup: ((bytes, bytes), int) tuple
+        :param message: Message with partition assigned.
```

Comment:
This docstring could use an update - something like "message with x, y, z assigned, ready to be enqueued"